astropy/io/fits/util.py

```diff
@@ -824,12 +824,24 @@ def _array_to_file_like(arr, fileobj):
     `numpy.ndarray.tofile`).
     """
 
+    if len(arr) == 0:
```

Comment:
According to PEP8 one generally uses `if not len(arr):` instead of directly comparing it with `== 0`.

As a side note: What does this line fix? Could you add a short comment why it is necessary or convenient?